### DISCUSSION TITLE
Downstream asset relied on Last Materialized Asset Partition

### DISCUSSION THREAD

**Discussion by steffyd** [dagster-io/dagster#18984] [Q&A] [answered]
I'm trying to have an asset rely on the upstream's asset latest materialized partition, and I'm struggling with figuring out how to determine that. I figured I could extend PartitionMapping but it doesn't appear that that would have any knowledge of the asset itself, just of the partition definition, which is asset agnostic.
If anyone has any advice, the use-case is this: We have an asset that gets updated annually, with an unknown cadence on when the data will be released. While we have use-cases for the historical data, we're currently trying to extract the latest materialization of the asset to perform a transformation. I opted to have the upstream asset be partitioned yearly, since this data is uploaded annually. However, with the new year, my downstream asset now expects us to use the 2023 partition, which doesn't currently exist as we're waiting for it to be released. How could I determine the latest materialized partition of my downstream asset? Or is this not possible with Dagster currently

**Comment by clairelin135** [comment, marked as answer]
Hi @steffyd.

We don't have a super ergonomic way to handle this use case currently. One workaround would be to define an `AllPartitionMapping` between the upstream asset and the downstream asset. This would require the downstream asset to load all upstream partitions. Then within the downstream asset's code you could fetch the latest partition with an existing output value.

Side note: unfortunately Dagster currently doesn't support defining custom `PartitionMapping`s. This is due to performance constraints when calling out to custom user code defined within `PartitionMapping`s.
